llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -470,6 +470,61 @@ def IREELinalgExt_ReverseOp : IREELinalgExt_Op<"reverse", [
   }];
 }
 
+def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
+  DeclareOpInterfaceMethods<LinalgExtInterface>
+]> {
+  let summary = "Top-K operator";
+  let description = [{
+   A Top-K operation for N-D tensors.
+
+   Accepts two N-D tensor inputs consisting of values (f32) and indices of those
+   values (i32 type). Both input values/indices tensors and output
+   values/indicies tensors must have the same shape. Top-K is computed along the
+   specified dimension. Returns two output tensors of values and the indicies of
+   Top-K results. The output dimensions must match the input save for the
+   dimension that is reduced to K results.
+
+   Setting "min_k" to true computes min K elements instead.
+  }];
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       I64Attr:$dimension,
+                       BoolAttr:$min_k
```

Comment:
I would rather add an operation with a region that represents a binary operation that encapsulates the comparison you need. Think of how std::sort works with "any sort" function. The same should be applicable here.